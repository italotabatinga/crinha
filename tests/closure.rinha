// outer closure from inside
let outer = fn () => {
  let x = "outside";
  let inner = fn () => {
    print x;
  };
  inner();
};
outer();

// outer closure from outside
let f = fn () => {
  let x = 1;
  fn () => {
    x
  }
};
let g = f();
print(g())

// local recursive function
let f = fn () => {
  let rec = fn () => {
    if (true) {
      1
    } else {
      rec();
    }
  };
  rec
};
let g = f();
print(g())