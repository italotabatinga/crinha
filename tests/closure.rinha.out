outside
1
1
